core/sr-primitives/src/traits.rs

```diff
@@ -50,47 +50,83 @@ impl<'a> Lazy<[u8]> for &'a [u8] {
 	fn get(&mut self) -> &[u8] { &**self }
 }
 
+/// Some type that is able to be collapsed into an account ID. It is not possible to recreate the original value from
+/// the account ID.
+pub trait IdentifyAccount {
+	/// The account ID that this can be transformed into.
+	type AccountId;
+	/// Transform into an account.
+	fn into_account(self) -> Self::AccountId;
+}
+
+impl IdentifyAccount for primitives::ed25519::Public {
+	type AccountId = Self;
+	fn into_account(self) -> Self { self }
+}
+
+impl IdentifyAccount for primitives::sr25519::Public {
+	type AccountId = Self;
+	fn into_account(self) -> Self { self }
+}
+
+impl IdentifyAccount for primitives::ecdsa::Public {
```

Comment:
It's weird that with the `MultiSigner` `into_account` hashes the key to 32 bytes whereas with this it does not. Why?

Reply:
Basically because the public key of the *25519 crypto is a straight 32-bytes, whereas the public key of SECP256k1 is an annoying 33 bytes. Since 32 bytes is a nice alignment that might have some good optimisations (e.g. for cache lines), and since `AccountId` is used copiously, I don't especially want to risk pessimising everything for the sake of supporting SECP256k1.

This basically just follows the pattern of Ethereum where the public key is hashed and that hash is used for the public account (though unlike Ethereum, we don't throw away 12 bytes of the hash).

For the *25519 crypto, it remains backwards compatible with Kusama.

Reply:
I was asking more why `primitives::ecdsa::public as IdentifyAccount` does not do the hashing, not  why `MultiSigner` does. I feel like it's be better if they were consistent with each other.

Reply:
`MultiSigner` is what forces the need for 32 byte `AccountId`s, given it's used to be backwards compatible with Kusama and the other two entries are naturally 32 bytes.